.github/workflows/publish.yaml

```diff
@@ -0,0 +1,73 @@
+name: Publish
+on:
+  push:
+    tags:
+      - '*'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      hash: ${{ steps.hash.outputs.hash }}
+    steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+        with:
+          python-version: '3.x'
+          cache: pip
+          cache-dependency-path: requirements*/*.txt
```

Comment:
Curious--is this what drove the use of `requirements` files?

Because they add some complexity here vs just doing `pip install -e .` or `pip install -e .[tests,docs]`

I'm actually surprised you didn't go that route with Flask...

Reply:
I don't like the using publicly installable extras for development only environments. It's also not possible to pin extras autoamtaically. pip-compile allows pinning multiple environments as well as automatic updates with Dependabot.